packages/alchemy/src/signer/session/manager.ts

```diff
@@ -1,7 +1,14 @@
+import EventEmitter from "eventemitter3";
 import { z } from "zod";
+import {
+  createJSONStorage,
+  persist,
+  subscribeWithSelector,
+} from "zustand/middleware";
+import { createStore, type Mutate, type StoreApi } from "zustand/vanilla";
```

Comment:
recoil is more tightly coupled with react, whereas zustand lets you define stores external to react. What that means is that I can create "events" which are actually subscriptions to the state changes. This also means that for our context provider, we can split apart the state that the context provider holds and the signer state. If you look at how wagmi does it, they have a config defined which actually holds the state. The provider context takes in a config and actually just provides that config to the rest of the app. This would let us build Embedded Account state that doesn't necessarily require react